### DISCUSSION TITLE
Any way to control the render fps for Avalonia?

### DISCUSSION THREAD

**Discussion by laolarou726** [AvaloniaUI/Avalonia#13654] [Q&A]
In WPF, you can do something like this to manually set the FPS for the application:

```c#
Timeline.DesiredFrameRateProperty.OverrideMetadata(typeof(Timeline),  
new FrameworkPropertyMetadata { DefaultValue = 30 });
```

Is there an equivalent thing in Avalonia?

**Comment by MinecraftYJQ** [comment]
It doesn't seem to be, but I haven't found it yet

**Reply by stevemonaco** [reply]
DevTools supports showing FPS as an overlay.

You can also enable the same overlay in your own app using the same approach DevTools uses:

https://github.com/AvaloniaUI/Avalonia/blob/46d4735868132e0eff094ee152f3d54517f01182/src/Avalonia.Diagnostics/Diagnostics/ViewModels/MainViewModel.cs#L87-L116

It may include some extra info you don't want in a real app. In that case, I'm not sure there's an approach besides using `RequestAnimationFrame`, `Render` override, etc and calculating+rendering FPS yourself.

**Reply by MinecraftYJQ** [reply]
I know. But I want to have a function directly to get it. and I wanted something that could directly control the FPS of the entire program.And the fetch you mentioned is only displayed on the screen, and only the debug mode is available

**Reply by stevemonaco** [reply]
You can enable the debug overlays even in release configuration builds. Being in charge of the FPS isn't really for the user app though it might be nice to have a low FPS mode for laptop/phone to save battery. That's the compositor's job that lives on its own render thread(s).
